src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -1963,7 +1963,48 @@ public TryWriteInterpolatedStringHandler(int literalLength, int formattedCount,
             /// <summary>Writes the specified string to the handler.</summary>
             /// <param name="value">The string to write.</param>
             /// <returns>true if the value could be formatted to the span; otherwise, false.</returns>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public bool AppendLiteral(string value)
+            {
+                // See comment on inlining and special-casing in DefaultInterpolatedStringHandler.AppendLiteral.
+
+                if (value.Length == 1)
+                {
+                    Span<char> destination = _destination;
+                    int pos = _pos;
+                    if ((uint)pos < (uint)destination.Length)
+                    {
+                        destination[pos] = value[0];
+                        _pos = pos + 1;
+                        return true;
+                    }
+
+                    return Fail();
+                }
+
+                if (value.Length == 2)
+                {
+                    Span<char> destination = _destination;
+                    int pos = _pos;
+                    if ((uint)pos < destination.Length - 1)
```

Comment:
why there is a cast here?

Reply:
If pos is negative, this would pass but then we'd access memory out of bounds.

Reply:
can it be negative though? we're tracking it ourselves

Reply:
No.  But the previous feedback was that we should be very cautious about it, anyway:
https://github.com/dotnet/runtime/pull/57217#discussion_r687001653